webapp/components/hooks/useInterceptBack.js

```diff
@@ -0,0 +1,39 @@
+import { useEffect } from 'react'
+
+/**
+ * Intercepts the browser back button press and calls the specified callback.
+ * The call to the callback can be can be activated or deactivated using the "active" parameter.
+ *
+ * @param {!object} params - The parameters.
+ * @param {!Function} [params.onBack] - Callback to be called when back button is pressed. It should return true when history.back() is called by the callback itself, false othwerwise.
+ * @param {boolean} [params.active=false] - True if onBack callback should be called on browser back button press and confirm message on page refresh should be shown.
+ * @returns {undefined}
+ */
+export const useInterceptBack = (params) => {
+  const { active = false, onBack } = params
+
+  const onBackButtonEvent = async (e) => {
+    e.preventDefault()
+
```

Comment:
why using window.history instead of react-router history ?

Reply:
how?

Reply:
useHistory ?

Reply:
useHistory... useHistoryListen... you can only listen to changes to the location. The best thing would be to avoid changes in the location, so you don't unmount/mount components.

Reply:
I meant to use history api https://reactrouter.com/web/api/history

Reply:
ok